filament/src/main/java/net/fabricmc/filament/FilamentGradlePlugin.java

```diff
@@ -1,25 +1,87 @@
+
+		TaskProvider<DownloadTask> minecraftClient = tasks.register("downloadMinecraftClientJar", DownloadTask.class, task -> {
+			Provider<MinecraftVersionMeta.Download> downloadProvider = metaProvider.map(meta -> meta.download("client"));
+			task.getUrl().set(downloadProvider.map(MinecraftVersionMeta.Download::url));
+			task.getSha1().set(downloadProvider.map(MinecraftVersionMeta.Download::sha1));
+
+			task.getOutputFile().set(new File(extension.getMinecraftDirectory(), "client.jar"));
+		});
+		TaskProvider<DownloadTask> minecraftServer = tasks.register("downloadMinecraftServerJar", DownloadTask.class, task -> {
+			Provider<MinecraftVersionMeta.Download> downloadProvider = metaProvider.map(meta -> meta.download("server"));
+			task.getUrl().set(downloadProvider.map(MinecraftVersionMeta.Download::url));
+			task.getSha1().set(downloadProvider.map(MinecraftVersionMeta.Download::sha1));
+
+			task.getOutputFile().set(new File(extension.getMinecraftDirectory(), "server_bundle.jar"));
+		});
+		TaskProvider<ExtractBundledServerTask> extractBundledServer = tasks.register("extractBundledServer", ExtractBundledServerTask.class, task -> {
+			task.dependsOn(minecraftServer);
+			task.getServerJar().set(getOutput(minecraftServer));
+			task.getOutputFile().set(new File(extension.getMinecraftDirectory(), "server.jar"));
+		});
+		tasks.register("mergeMinecraftJars", MergeMinecraftTask.class, task -> {
+			task.getClientJar().set(getOutput(minecraftClient));
+			task.getServerJar().set(getOutput(extractBundledServer));
 
-		var combineUnpickDefinitions = project.getTasks().register("combineUnpickDefinitions", CombineUnpickDefinitionsTask.class);
-		project.getTasks().register("remapUnpickDefinitionsIntermediary", RemapUnpickDefinitionsTask.class, task -> {
+			task.getOutputFile().set(new File(extension.getMinecraftDirectory(), "merged.jar"));
+		});
+		tasks.register("generatePackageInfoMappings", GeneratePackageInfoMappingsTask.class);
+		tasks.register("javadocLint", JavadocLintTask.class);
+
+		TaskProvider<CombineUnpickDefinitionsTask> combineUnpickDefinitions = tasks.register("combineUnpickDefinitions", CombineUnpickDefinitionsTask.class);
+		tasks.register("remapUnpickDefinitionsIntermediary", RemapUnpickDefinitionsTask.class, task -> {
 			task.dependsOn(combineUnpickDefinitions);
 			task.getInput().set(combineUnpickDefinitions.flatMap(CombineUnpickDefinitionsTask::getOutput));
 			task.getSourceNamespace().set("named");
 			task.getTargetNamespace().set("intermediary");
 		});
+
+		project.getConfigurations().register("minecraftLibraries", configuration -> configuration.setTransitive(false));
+
+		GradleUtils.afterSuccessfulEvaluation(project, () -> afterEvaluate(project));
+	}
+
+	private void afterEvaluate(Project project) {
+		try {
+			MinecraftMetadata.setup(project);
+		} catch (Exception e) {
+			throw new RuntimeException("Failed to setup", e);
+		}
+	}
```

Comment:
Can we do it in tasks here? I think that works better for Yarn where you don't have an MC dev env (and doesn't require aE)

Reply:
Ill try again, I was struggling with having none File based project outputs. The solution may be to just re-read the metadata each time its needed.